incubator/postfix/templates/secrets.yaml

```diff
@@ -9,6 +9,11 @@ metadata:
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
-  smtp_username: {{ trim .Values.smtp.username | b64enc }}
   smtp_password : {{ trim .Values.smtp.password | b64enc }}
   smtp_domain: {{ trim .Values.smtp.domain | b64enc }}
```

Comment:
My point is that we shouldn't define the `smtp_auth_*` vars if they are not enabled because we don't define the `SMTP_AUTH_*` env if they are not enabled.